### ISSUE TITLE
configure configures wrong mpi path and says... yes

### ISSUE THREAD

**Issue by ggruszczynski** [CFD-GO/TCLB#306]
I found that that configure output is misleading.
`checking for mpi.h... yes` 
shall say `oups`

$ ./configure --enable-double --enable-graphics --with-cuda-arch=sm_60
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether mpic++ accepts -g... yes
checking for gcc... mpicc
checking whether we are using the GNU C compiler... yes
checking whether mpicc accepts -g... yes
checking for mpicc option to accept ISO C89... none needed
checking how to run the C++ preprocessor... mpic++ -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking float.h usability... yes
checking float.h presence... yes
checking for float.h... yes
checking malloc.h usability... yes
checking malloc.h presence... yes
checking for malloc.h... yes
checking stddef.h usability... yes
checking stddef.h presence... yes
checking for stddef.h... yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking wchar.h usability... yes
checking wchar.h presence... yes
checking for wchar.h... yes
checking for sqrt in -lm... yes
checking CUDA home directory... checking for nvcc... yes
using /usr/local/cuda-11.0
checking for /usr/local/cuda-11.0/bin/nvcc... yes
checking for /usr/local/cuda-11.0/lib64/libcudart.so... yes
using CUDALIBDIR:/usr/local/cuda-11.0/lib64
checking cuda.h usability... yes
checking cuda.h presence... yes
checking for cuda.h... yes
checking GL/glut.h usability... yes
checking GL/glut.h presence... yes
checking for GL/glut.h... yes
checking for glutInit in -lglut... yes
checking for glDrawPixels in -lGL... yes
checking for gluSphere in -lGLU... yes
checking target CUDA architecture... sm_60
**checking MPI include path... /usr/include/mpi
checking MPI library path... /usr/lib/openmpi/lib
checking mpi.h usability... yes
checking mpi.h presence... yes
checking for mpi.h... yes**
checking for MPI_Recv in -lmpi... yes
checking HDF5 path... checking hdf5.h usability... no
checking hdf5.h presence... no
checking for hdf5.h... no
checking for H5Pset_fapl_mpio in -lhdf5... no
checking for local NLOpt installation... not found
checking nlopt.h usability... no
checking nlopt.h presence... no
checking for nlopt.h... no
checking for nlopt_create in -lnlopt... no
checking for R... /usr/bin/R
checking /usr/bin/R... exists
checking R package: optparse... yes
checking R package: numbers... yes
checking R package: rtemplate... yes
checking R package: gvector... yes
checking R package: polyAlgebra... yes
checking R package: RInside... no
checking if RInside required... no
configure: creating ./config.status
config.status: creating CLB/config.mk
config.status: creating config.main.mk
config.status: creating CLB/config.h
config.status: CLB/config.h is unchanged

$ make -j 6 d2q9
  RT         makefile.main
  RT         CLB/d2q9/Global.cpp
  MKDIR      CLB/d2q9/
  RT         CLB/d2q9/Lattice.cu
  RT         CLB/d2q9/vtkLattice.cpp
  CP         CLB/d2q9/vtkOutput.cpp
  CP         CLB/d2q9/cross.cu
  RT         CLB/d2q9/cuda.cu
  RT         CLB/d2q9/LatticeContainer.inc.cpp
  RT         CLB/d2q9/LatticeAccess.inc.cpp
  RT         CLB/d2q9/Dynamics.c (model)
  RT         CLB/d2q9/Solver.cpp
  CP         CLB/d2q9/pugixml.cpp
  RT         CLB/d2q9/Geometry.cpp
  RT         CLB/d2q9/def.cpp
  CP         CLB/d2q9/unit.cpp
  CP         CLB/d2q9/ZoneSettings.cpp
  CP         CLB/d2q9/SyntheticTurbulence.cpp
  RT         CLB/d2q9/Sampler.cpp
  CP         CLB/d2q9/main.cpp
  RT         CLB/d2q9/Global.h
  CP         CLB/d2q9/gpu_anim.h
  RT         CLB/d2q9/LatticeContainer.h
  RT         CLB/d2q9/Lattice.h
  CP         CLB/d2q9/Region.h
  CP         CLB/d2q9/vtkLattice.h
  CP         CLB/d2q9/vtkOutput.h
  CP         CLB/d2q9/cross.h
  CP         CLB/d2q9/cross.hpp
  CP         CLB/d2q9/gl_helper.h
  RT         CLB/d2q9/Dynamics.h
  CP         CLB/d2q9/types.h
  RT         CLB/d2q9/Consts.h
  RT         CLB/d2q9/Solver.h
  CP         CLB/d2q9/pugixml.hpp
  CP         CLB/d2q9/pugiconfig.hpp
  CP         CLB/d2q9/Geometry.h
  CP         CLB/d2q9/def.h
  CP         CLB/d2q9/utils.h
  CP         CLB/d2q9/unit.h
  CP         CLB/d2q9/ZoneSettings.h
  CP         CLB/d2q9/SyntheticTurbulence.h
  CP         CLB/d2q9/Sampler.h
  CP         CLB/d2q9/spline.h
  CP         CLB/d2q9/TCLBForceGroupCommon.h
  CP         CLB/d2q9/RemoteForceInterface.cpp
  CP         CLB/d2q9/RemoteForceInterface.h
  CP         CLB/d2q9/RemoteForceInterface.hpp
  CP         CLB/d2q9/MPMD.hpp
  CP         CLB/d2q9/empty.cpp
  CP         CLB/d2q9/Particle.hpp
  CP         CLB/d2q9/lammps.cpp
  CP         CLB/d2q9/BallTree.h
  CP         CLB/d2q9/BallTree.hpp
  CP         CLB/d2q9/BallTree.cpp
  CP         CLB/d2q9/xpath_modification.cpp
  CP         CLB/d2q9/xpath_modification.h
  RT         CLB/d2q9/hdf5Lattice.cpp
  CP         CLB/d2q9/hdf5Lattice.h
  CP         CLB/d2q9/glue.hpp
  CP         CLB/d2q9/mpitools.hpp
  CP         CLB/d2q9/pinned_allocator.hpp
  CP         CLB/d2q9/compare.cpp
  CP         CLB/d2q9/simplepart.cpp
  CP         CLB/d2q9/Handlers/acAndersen.cpp
  MKDIR      CLB/d2q9/Handlers/
  CP         CLB/d2q9/Handlers/acAndersen.h
  CP         CLB/d2q9/Handlers/acContainer.cpp
  CP         CLB/d2q9/Handlers/acContainer.h
  CP         CLB/d2q9/Handlers/acESYSParticle.cpp
  CP         CLB/d2q9/Handlers/acESYSParticle.h
  CP         CLB/d2q9/Handlers/acFDTest.cpp
  CP         CLB/d2q9/Handlers/acFDTest.h
  CP         CLB/d2q9/Handlers/acGeometry.cpp
  CP         CLB/d2q9/Handlers/acGeometry.h
  CP         CLB/d2q9/Handlers/acInit.cpp
  CP         CLB/d2q9/Handlers/acInit.h
  CP         CLB/d2q9/Handlers/acLoadBinary.cpp
  CP         CLB/d2q9/Handlers/acLoadBinary.h
  CP         CLB/d2q9/Handlers/acLoadMemoryDump.cpp
  CP         CLB/d2q9/Handlers/acLoadMemoryDump.h
  CP         CLB/d2q9/Handlers/acModel.cpp
  CP         CLB/d2q9/Handlers/acModel.h
  RT         CLB/d2q9/Handlers/acObjective.cpp
  CP         CLB/d2q9/Handlers/acObjective.h
  RT         CLB/d2q9/Handlers/acOptimize.cpp
  CP         CLB/d2q9/Handlers/acOptimize.h
  CP         CLB/d2q9/Handlers/acOptSolve.cpp
  CP         CLB/d2q9/Handlers/acOptSolve.h
  RT         CLB/d2q9/Handlers/acParam.cpp
  CP         CLB/d2q9/Handlers/acParam.h
  CP         CLB/d2q9/Handlers/acRemoteForceInterface.cpp
  CP         CLB/d2q9/Handlers/acRemoteForceInterface.h
  CP         CLB/d2q9/Handlers/acRepeat.cpp
  CP         CLB/d2q9/Handlers/acRepeat.h
  RT         CLB/d2q9/Handlers/acRunAction.cpp
Warning message:
In readLines(f) : incomplete final line found on 'src/hdf5Lattice.cpp.Rt'
  CP         CLB/d2q9/Handlers/acRunAction.h
  CP         CLB/d2q9/Handlers/acSAdjoint.cpp
  CP         CLB/d2q9/Handlers/acSAdjoint.h
  CP         CLB/d2q9/Handlers/acSolve.cpp
  CP         CLB/d2q9/Handlers/acSolve.h
  CP         CLB/d2q9/Handlers/acSyntheticTurbulence.cpp
  CP         CLB/d2q9/Handlers/acSyntheticTurbulence.h
  RT         CLB/d2q9/Handlers/acThreshold.cpp
  CP         CLB/d2q9/Handlers/acThreshold.h
  RT         CLB/d2q9/Handlers/acThresholdNow.cpp
  CP         CLB/d2q9/Handlers/acThresholdNow.h
  CP         CLB/d2q9/Handlers/Action.cpp
  CP         CLB/d2q9/Handlers/Action.h
  CP         CLB/d2q9/Handlers/acUSAdjoint.cpp
  CP         CLB/d2q9/Handlers/acUSAdjoint.h
  CP         CLB/d2q9/Handlers/BSpline.cpp
  CP         CLB/d2q9/Handlers/BSpline.h
  CP         CLB/d2q9/Handlers/Callback.cpp
  CP         CLB/d2q9/Handlers/Callback.h
  CP         CLB/d2q9/Handlers/cbAveraging.cpp
  CP         CLB/d2q9/Handlers/cbAveraging.h
  CP         CLB/d2q9/Handlers/cbBIN.cpp
  CP         CLB/d2q9/Handlers/cbBIN.h
  CP         CLB/d2q9/Handlers/cbCatalyst.cpp
  CP         CLB/d2q9/Handlers/cbCatalyst.h
  CP         CLB/d2q9/Handlers/cbDumpSettings.cpp
  CP         CLB/d2q9/Handlers/cbDumpSettings.h
  RT         CLB/d2q9/Handlers/cbFailcheck.cpp
  CP         CLB/d2q9/Handlers/cbFailcheck.h
  CP         CLB/d2q9/Handlers/cbHDF5.cpp
  CP         CLB/d2q9/Handlers/cbHDF5.h
  CP         CLB/d2q9/Handlers/cbKeep.cpp
  CP         CLB/d2q9/Handlers/cbKeep.h
  CP         CLB/d2q9/Handlers/cbLog.cpp
  CP         CLB/d2q9/Handlers/cbLog.h
  RT         CLB/d2q9/Handlers/cbPID.cpp
  CP         CLB/d2q9/Handlers/cbPID.h
  CP         CLB/d2q9/Handlers/cbPythonCall.cpp
  CP         CLB/d2q9/Handlers/cbPythonCall.h
  RT         CLB/d2q9/Handlers/cbRunR.cpp
  CP         CLB/d2q9/Handlers/cbRunR.h
  CP         CLB/d2q9/Handlers/cbSample.cpp
  CP         CLB/d2q9/Handlers/cbSample.h
  CP         CLB/d2q9/Handlers/cbSaveBinary.cpp
  CP         CLB/d2q9/Handlers/cbSaveBinary.h
  CP         CLB/d2q9/Handlers/cbSaveMemoryDump.cpp
  CP         CLB/d2q9/Handlers/cbSaveMemoryDump.h
  RT         CLB/d2q9/Handlers/cbStop.cpp
  CP         CLB/d2q9/Handlers/cbStop.h
  CP         CLB/d2q9/Handlers/cbTXT.cpp
  CP         CLB/d2q9/Handlers/cbTXT.h
  CP         CLB/d2q9/Handlers/cbVTK.cpp
  CP         CLB/d2q9/Handlers/cbVTK.h
  RT         CLB/d2q9/Handlers/conControl.cpp
  CP         CLB/d2q9/Handlers/conControl.h
  CP         CLB/d2q9/Handlers/conControlParameter.cpp
  CP         CLB/d2q9/Handlers/conControlParameter.h
  CP         CLB/d2q9/Handlers/conExtrude.cpp
  CP         CLB/d2q9/Handlers/conExtrude.h
  RT         CLB/d2q9/Handlers/conFieldParameter.cpp
  CP         CLB/d2q9/Handlers/conFieldParameter.h
  CP         CLB/d2q9/Handlers/Design.cpp
  CP         CLB/d2q9/Handlers/Design.h
  CP         CLB/d2q9/Handlers/Fourier.cpp
  CP         CLB/d2q9/Handlers/Fourier.h
  CP         CLB/d2q9/Handlers/GenericAction.cpp
  CP         CLB/d2q9/Handlers/GenericAction.h
  CP         CLB/d2q9/Handlers/GenericContainer.cpp
  CP         CLB/d2q9/Handlers/GenericContainer.h
  CP         CLB/d2q9/Handlers/GenericOptimizer.cpp
  CP         CLB/d2q9/Handlers/GenericOptimizer.h
  RT         CLB/d2q9/Handlers/IfContainer.cpp
  RT         CLB/d2q9/Handlers/IfContainer.h
  CP         CLB/d2q9/Handlers/InternalTopology.cpp
  CP         CLB/d2q9/Handlers/InternalTopology.h
  CP         CLB/d2q9/Handlers/MainContainer.cpp
  CP         CLB/d2q9/Handlers/MainContainer.h
  CP         CLB/d2q9/Handlers/NullHandler.cpp
  CP         CLB/d2q9/Handlers/NullHandler.h
  RT         CLB/d2q9/Handlers/OptimalControl.cpp
  CP         CLB/d2q9/Handlers/OptimalControl.h
  RT         CLB/d2q9/Handlers/OptimalControlSecond.cpp
  CP         CLB/d2q9/Handlers/OptimalControlSecond.h
  CP         CLB/d2q9/Handlers/RepeatControl.cpp
  CP         CLB/d2q9/Handlers/RepeatControl.h
  CP         CLB/d2q9/Handlers/vHandler.cpp
  CP         CLB/d2q9/Handlers/vHandler.h
  CP         CLB/d2q9/CommonHandler.h
  CP         CLB/d2q9/Handlers.h
  CP         CLB/d2q9/Factory.h
  CP         CLB/d2q9/HandlerFactory.h
  RT         CLB/d2q9/SUMMARY
  CP         CLB/d2q9/conf.doxygen
  RT         CLB/d2q9/Catalyst.cpp
make: Circular CLB/d2q9/makefile <- CLB/d2q9/makefile dependency dropped.
  CP         CLB/d2q9/Catalyst.h
  CP         CLB/d2q9//README.md
  RT         CLB/d2q9//dictionary.h
  SP-CONST   CLB/d2q9/Dynamics_sp.c
  AUTO-DEP   CLB/d2q9/dep.mk
  RT         CLB/d2q9/makefile
make[1]: Entering directory '/home/grzegorz/GITHUB/LBM/TCLB/CLB/d2q9'
  G++        empty.cpp
  G++        compare.cpp
  G++        Global.cpp
  G++        pugixml.cpp
  G++        def.cpp
  G++        ZoneSettings.cpp
  G++        RemoteForceInterface.cpp
  G++        xpath_modification.cpp
  G++        simplepart.cpp
  G++        vtkOutput.cpp
  G++        unit.cpp
  G++        SyntheticTurbulence.cpp
  G++        Handlers/NullHandler.cpp
  LINKING    empty
  NVCC       cross.cu
  G++        Geometry.cpp
In file included from Global.h:26,
                 from cross.cu:4:
MPMD.hpp:9:10: fatal error: mpi.h: No such file or directory
    9 | #include <mpi.h>
      |          ^~~~~~~
compilation terminated.
make[1]: *** [makefile:63: cross.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/grzegorz/GITHUB/LBM/TCLB/CLB/d2q9'
make: *** [makefile.main:16471: CLB/d2q9/main] Error 2

```
$ whereis mpi
mpi: /usr/lib/x86_64-linux-gnu/openmpi/include/mpi.h
```
```
$ mpicc -showme 
gcc -I/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi -I/usr/lib/x86_64-linux-gnu/openmpi/include -pthread -L/usr/lib/x86_64-linux-gnu/openmpi/lib -lmpi
```
```
$ locate mpi.h
/usr/lib/x86_64-linux-gnu/openmpi/include/mpi.h
/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/fortran/mpif-h/prototypes_mpi.h
/usr/src/linux-headers-5.4.0-42/include/linux/mpi.h
```

```
$ mpirun --version
mpirun (Open MPI) 4.0.3
```
```
$ mpic++ --version
g++ (Ubuntu 9.3.0-10ubuntu2) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
$ mpicc --version
gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

-----------------------------
**in CLB/conf.mk**

Shall be
```
OPT =  -I/usr/local/cuda-11.0/include -I/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi -I/usr/lib/x86_64-linux-gnu/openmpi/include -pthread -O3 -Wno-write-strings
CPP_OPT =  -std=c++11 
LD_OPT =  -L/usr/local/cuda-11.0/lib64 -L/usr/lib/x86_64-linux-gnu/openmpi/lib -lmpi_cxx -lmpi -lGLU -lGL -lglut -lm 
NV_OPT = -D_FORCE_INLINES   -arch=compute_60 -std=c++11
CPU = 
```

Was
```
OPT =  -I/usr/local/cuda-11.0/include -I/usr/include/mpi -O3 -Wno-write-strings
CPP_OPT =  -std=c++11
LD_OPT =  -L/usr/local/cuda-11.0/lib64 -L/usr/lib/openmpi/lib -lmpi -lGLU -lGL -lglut -lm 
NV_OPT = -D_FORCE_INLINES   -arch=compute_60 -std=c++11
CPU = 
```

Suprisingly it was happy to compile on cpu, `./configure --enable-double --enable-cpp11  --disable-cuda 
` then the automatically generated `CLB/conf.mk` was

```
OPT =  -I/usr/include/mpi -O3 -Wno-write-strings
CPP_OPT =  -std=c++11
LD_OPT =  -L/usr/lib/openmpi/lib -lmpi -lm 
NV_OPT = -D_FORCE_INLINES   -std=c++11
CPU = 1
```

I guess that the error origins in https://github.com/CFD-GO/TCLB/blob/master/src/configure.ac

**Comment by llaniewski** [comment]
But is there a `mpi.h` in '/usr/include/mpi/` ?

**Comment by ggruszczynski** [comment]
```
$ ls /usr/include/mpi/
ls: cannot access '/usr/include/mpi/': No such file or directory
```

**Comment by llaniewski** [comment]
Reproduced on Ubuntu 20.04.1 LTS. The error is raised only by `nvcc`.

**Comment by llaniewski** [comment]
Tried fixing the issue in #325 but apparently it broke something on Travis-CI. Will get back to it.

**Comment by ggruszczynski** [comment]
> Reproduced on Ubuntu 20.04.1 LTS. The error is raised only by `nvcc`.

In my case, it also happened on Ubuntu 20.04.

**Comment by llaniewski** [comment]
Fixed in #325
